src/Compilers/CSharp/Portable/Compilation/CSharpCompilation.cs

```diff
@@ -2051,14 +2051,15 @@ internal override bool FilterAndAppendAndFreeDiagnostics(DiagnosticBag accumulat
         /// Filter out warnings based on the compiler options (/nowarn, /warn and /warnaserror) and the pragma warning directives.
         /// </summary>
         /// <returns>True when there is no error.</returns>
-        private bool FilterAndAppendDiagnostics(DiagnosticBag accumulator, IEnumerable<Diagnostic> incoming)
+        private bool FilterAndAppendDiagnostics(DiagnosticBag accumulator, IEnumerable<Diagnostic> incoming, bool includeDiagnosticsWithSourceSuppression = false)
         {
             bool hasError = false;
 
             foreach (Diagnostic d in incoming)
             {
                 var filtered = _options.FilterDiagnostic(d);
-                if (filtered == null)
+                if (filtered == null ||
+                    (!includeDiagnosticsWithSourceSuppression && filtered.HasSourceSuppression))
```

Comment:
Yes you are correct. The feature enhancement here is just to surface the diagnostics with suppressions in source, not those suppressed via CompilationOptions (nowarn, ruleset, etc.).